### PR TITLE
NIFI-14064: Descriptions are not shown for Python processors

### PR DESCRIPTION
**Pull request by takraj**
Bug's been present since fb2a82c035 / PR #9288 , which changed the way documentations are generated. The new view dyamically fetches the information to be shown from the NIFI API, which did not return the component tags for Python processors in the manifest. Since the new documentation viewer always expects the tags to be returned, it has errored out when a python component was selected, which resulted in none of the documentation had been shown.

This PR attempts to fix this issue by adding the missing information to the API response.

# Summary

[NIFI-14064](https://issues.apache.org/jira/browse/NIFI-14064)

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [x] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [x] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [x] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [x] Pull Request based on current revision of the `main` branch
- [x] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [x] Build completed using `mvn clean install -P contrib-check`
  - [x] JDK 21

### Licensing

- [x] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [x] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [x] Documentation formatting appears as expected in rendered files

### CODE REVIEW

File: nifi-extension-bundles/nifi-py4j-extension-bundle/nifi-py4j-integration-tests/src/test/java/org.apache.nifi.py4j/PythonControllerInteractionIT.java [hunk 1]
```diff
@@ -17,6 +17,28 @@
...
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
```
Comment (exceptionfactory) [inline nifi-extension-bundles/nifi-py4j-extension-bundle/nifi-py4j-integration-tests/src/test/java/org.apache.nifi.py4j/PythonControllerInteractionIT.java:41]: The import ordering appears to be the result of auto-formatting, please revert these changes.

**Review by exceptionfactory — CHANGES_REQUESTED** [review]
@takraj Can you clarify and add some details about the changes in this pull request? The only runtime code change is the addition of Tags from Python Processors, but the title of the pull request says Descriptions are not shown. The description does appear to be set already in `StandardRuntimeManifestService.createExtension()`.

The test updates appear to be useful, but given the scope of the changes, it would be helpful to revert the import ordering changes.

**Comment by takraj** [comment]
@exceptionfactory Sure. The referenced PR https://github.com/apache/nifi/pull/9288 changes the way component documentations are generated. From my understanding, earlier these documentations were somehow pre-generated during startup, but now it is loaded dynamically by the documentation view, using the NIFI REST API. Since this PR, when I add a Python processor, and open the doc view, basically only the name, version, and the relationships are shown. On the other hand, it works as expected for normal Java based processors. When I opened the browser's debug console I've spotted some error messages, when selecting any of the Python processors, that were referring to the missing 'tags' key from the API response. I have checked the endpoint manually, and compared the responses for Python and Java processors, and confirmed that some of the keys are missing in the former case. The Python processor had tags defined as well, but they were missing from the API response. And the rest is basically tracing back the code...

File: nifi-extension-bundles/nifi-py4j-extension-bundle/nifi-py4j-integration-tests/src/test/java/org.apache.nifi.py4j/PythonControllerInteractionIT.java (hunk 1 above)
Comment (takraj) [reply nifi-extension-bundles/nifi-py4j-extension-bundle/nifi-py4j-integration-tests/src/test/java/org.apache.nifi.py4j/PythonControllerInteractionIT.java]: Addressed in a27a88b

File: nifi-framework-bundle/nifi-framework/nifi-framework-core/src/test/java/org/apache/nifi/manifest/StandardRuntimeManifestServiceTest.java [hunk 2]
```diff
@@ -25,15 +25,22 @@
...
 import org.apache.nifi.extension.manifest.parser.jaxb.JAXBExtensionManifestParser;
+import org.apache.nifi.nar.ExtensionDefinition;
+import org.apache.nifi.nar.ExtensionDefinition.ExtensionRuntime;
 import org.apache.nifi.nar.ExtensionManager;
+import org.apache.nifi.nar.PythonBundle;
+import org.apache.nifi.processor.Processor;
+import org.apache.nifi.python.PythonProcessorDetails;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+import org.mockito.internal.util.collections.Sets;
```
Comment (exceptionfactory) [inline nifi-framework-bundle/nifi-framework/nifi-framework-core/src/test/java/org/apache/nifi/manifest/StandardRuntimeManifestServiceTest.java:36]: Internal Mockito utility classes should never be used. It looks like this can be replaced with `Set.of()`

File: nifi-extension-bundles/nifi-py4j-extension-bundle/nifi-py4j-integration-tests/src/test/java/org.apache.nifi.py4j/PythonControllerInteractionIT.java [hunk 3]
```diff
@@ -160,6 +161,7 @@ public void testGetProcessorDetails() {
             .orElseThrow(() -> new RuntimeException("Could not find ConvertCsvToExcel"));
 
         assertEquals("0.0.1-SNAPSHOT", convertCsvToExcel.getProcessorVersion());
+        assertEquals(Arrays.asList("csv", "excel"), convertCsvToExcel.getTags());
```
Comment (exceptionfactory) [inline nifi-extension-bundles/nifi-py4j-extension-bundle/nifi-py4j-integration-tests/src/test/java/org.apache.nifi.py4j/PythonControllerInteractionIT.java:164]:
```suggestion
        assertEquals(List.of("csv", "excel"), convertCsvToExcel.getTags());
```

File: nifi-framework-bundle/nifi-framework/nifi-framework-core/src/test/java/org/apache/nifi/manifest/StandardRuntimeManifestServiceTest.java [hunk 4]
```diff
@@ -105,6 +120,49 @@ public void testGetRuntimeManifest() {
         assertEquals(1, controllerServiceDefinitions.size());
     }
 
+    @Test
+    public void testGetPythonManifest() {
+        final String CLASS_NAME = "ClassName";
+        final List<String> EXPECTED_TAGS = Arrays.asList("tag1", "tag2");
```
Comment (exceptionfactory) [inline nifi-framework-bundle/nifi-framework/nifi-framework-core/src/test/java/org/apache/nifi/manifest/StandardRuntimeManifestServiceTest.java:126]:
```suggestion
        final List<String> EXPECTED_TAGS = List.of("tag1", "tag2");
```

File: nifi-framework-bundle/nifi-framework/nifi-framework-core/src/test/java/org/apache/nifi/manifest/StandardRuntimeManifestServiceTest.java [hunk 5]
```diff
@@ -105,6 +120,49 @@ public void testGetRuntimeManifest() {
...
+
+        when(extensionManager.getAllBundles()).thenReturn(emptySet());
+        when(extensionManager.getExtensions(Processor.class)).thenReturn(Sets.newSet(
```
Comment (exceptionfactory) [inline nifi-framework-bundle/nifi-framework/nifi-framework-core/src/test/java/org/apache/nifi/manifest/StandardRuntimeManifestServiceTest.java:129]:
```suggestion
        when(extensionManager.getExtensions(Processor.class)).thenReturn(Set.of(
```

**Review by exceptionfactory — CHANGES_REQUESTED** [review]
Thanks for the additional background @takraj, that is helpful. For future clarity, summarizing the changes in the PR title and description, as opposed to just mentioning the behavioral bug, would be helpful.

Thanks for reverting the import ordering changes. I noted a few additional adjustments necessary regarding use of some collection classes, and otherwise this looks close to completion.

File: nifi-framework-bundle/nifi-framework/nifi-framework-core/src/test/java/org/apache/nifi/manifest/StandardRuntimeManifestServiceTest.java (hunk 2 above)
Comment (takraj) [reply nifi-framework-bundle/nifi-framework/nifi-framework-core/src/test/java/org/apache/nifi/manifest/StandardRuntimeManifestServiceTest.java]: Addressed in  030405d

File: nifi-extension-bundles/nifi-py4j-extension-bundle/nifi-py4j-integration-tests/src/test/java/org.apache.nifi.py4j/PythonControllerInteractionIT.java (hunk 3 above)
Comment (takraj) [reply nifi-extension-bundles/nifi-py4j-extension-bundle/nifi-py4j-integration-tests/src/test/java/org.apache.nifi.py4j/PythonControllerInteractionIT.java]: Addressed in  030405d

File: nifi-framework-bundle/nifi-framework/nifi-framework-core/src/test/java/org/apache/nifi/manifest/StandardRuntimeManifestServiceTest.java (hunk 4 above)
Comment (takraj) [reply nifi-framework-bundle/nifi-framework/nifi-framework-core/src/test/java/org/apache/nifi/manifest/StandardRuntimeManifestServiceTest.java]: Addressed in  030405d

File: nifi-framework-bundle/nifi-framework/nifi-framework-core/src/test/java/org/apache/nifi/manifest/StandardRuntimeManifestServiceTest.java (hunk 5 above)
Comment (takraj) [reply nifi-framework-bundle/nifi-framework/nifi-framework-core/src/test/java/org/apache/nifi/manifest/StandardRuntimeManifestServiceTest.java]: Addressed in  030405d

**Comment by takraj** [comment]
@exceptionfactory I have updated the PR description to reflect my intent.

**Review by exceptionfactory — APPROVED** [review]
Thanks for making the adjustments and clarifications @takraj, looks good! +1 merging
